lib/compute_visible_services.ts

```diff
@@ -0,0 +1,165 @@
+import * as winston from "winston";
+import * as documentDbUtils from "../lib/utils/documentdb";
+
+import { IContext } from "azure-function-express";
+import { DocumentClient } from "documentdb";
+
+import { pick } from "italia-ts-commons/lib/types";
+
+import { insert, lookup, remove, StrMap } from "fp-ts/lib/StrMap";
+
+import {
+  RetrievedService,
+  SERVICE_COLLECTION_NAME,
+  ServiceModel
+} from "./models/service";
+import { getRequiredStringEnv } from "./utils/env";
+
+import { createBlobService } from "azure-storage";
+import { isLeft } from "fp-ts/lib/Either";
+import { isNone, isSome } from "fp-ts/lib/Option";
+import {
+  VISIBLE_SERVICE_BLOB_ID,
+  VISIBLE_SERVICE_CONTAINER,
+  VisibleService
+} from "./models/visible_service";
+import { upsertBlobFromObject } from "./utils/azure_storage";
+import { configureAzureContextTransport } from "./utils/logging";
+
+// Whether we're in a production environment
+const isProduction = process.env.NODE_ENV === "production";
+
+const cosmosDbUri = getRequiredStringEnv("CUSTOMCONNSTR_COSMOSDB_URI");
+const cosmosDbKey = getRequiredStringEnv("CUSTOMCONNSTR_COSMOSDB_KEY");
+const cosmosDbName = getRequiredStringEnv("COSMOSDB_NAME");
+
+const documentDbDatabaseUrl = documentDbUtils.getDatabaseUri(cosmosDbName);
+
+const servicesCollectionUrl = documentDbUtils.getCollectionUri(
+  documentDbDatabaseUrl,
+  SERVICE_COLLECTION_NAME
+);
+const documentClient = new DocumentClient(cosmosDbUri, {
+  masterKey: cosmosDbKey
+});
+
+const serviceModel = new ServiceModel(documentClient, servicesCollectionUrl);
+
+const storageConnectionString = getRequiredStringEnv("QueueStorageConnection");
+const blobService = createBlobService(storageConnectionString);
+
+/**
+ * A function to cache the list of visible (active) services.
+ *
+ * Scheduled using a timer trigger, see
+ * https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/azure-functions/functions-bindings-timer.md
+ */
+export async function index(context: IContext<{}>): Promise<void> {
+  const logLevel = isProduction ? "info" : "debug";
+  configureAzureContextTransport(context, winston, logLevel);
+  try {
+    const servicesCollectionIterator = await serviceModel.getCollectionIterator();
+    // iterate over the whole services collection and collect visible services
+    const servicesIterator = documentDbUtils.reduceResultIterator(
+      servicesCollectionIterator,
+      (
```

Comment:
to make the code more readable, could you declare this function outside `index` and reference to it by name?